tensorflow/core/kernels/data/repeat_dataset_op_test.cc

```diff
@@ -142,26 +141,26 @@ class RepeatDatasetOpTestHelper : public RepeatDatasetOpTest {
     Tensor count = CreateTensor<int64>(TensorShape{}, {test_case.count});
     inputs.emplace_back(&tensor_slice_dataset_tensor);
     inputs.emplace_back(&count);
+    std::unique_ptr<OpKernel> dataset_kernel;
     TF_RETURN_IF_ERROR(CreateRepeatDatasetKernel(
         test_case.expected_output_dtypes, test_case.expected_output_shapes,
-        &dataset_kernel_));
-    TF_RETURN_IF_ERROR(CreateRepeatDatasetContext(
-        dataset_kernel_.get(), &inputs, &dataset_kernel_ctx_));
-    TF_RETURN_IF_ERROR(CreateDataset(dataset_kernel_.get(),
+        &dataset_kernel));
+    TF_RETURN_IF_ERROR(CreateRepeatDatasetContext(dataset_kernel.get(), &inputs,
+                                                  &dataset_kernel_ctx_));
+    TF_RETURN_IF_ERROR(CreateDataset(dataset_kernel.get(),
                                      dataset_kernel_ctx_.get(), &dataset_));
     return Status::OK();
   }
 
-  Status CreateIteratorFromTestCase(const TestParam &test_case) {
-    TF_RETURN_IF_ERROR(CreateDatasetFromTestCase(test_case));
+  Status CreateIterator(const TestCase &test_case) {
+    TF_RETURN_IF_ERROR(CreateDataset(test_case));
     TF_RETURN_IF_ERROR(
         CreateIteratorContext(dataset_kernel_ctx_.get(), &iterator_ctx_));
     TF_RETURN_IF_ERROR(
         dataset_->MakeIterator(iterator_ctx_.get(), "Iterator", &iterator_));
     return Status::OK();
   }
 
-  std::unique_ptr<OpKernel> dataset_kernel_;
   std::unique_ptr<OpKernelContext> dataset_kernel_ctx_;
```

Comment:
My comment was meant for all of the internal state (`dataset_kernel_`, `dataset_kernel_ctx_`, and `iterator_ctx_`).

While writing the code this way makes it less verbose, it makes it harder to understand what is going on in case someone would like to modify the code.